flowmachine/flowmachine/features/subscriber/topup_stats.py

```diff
@@ -0,0 +1,325 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# -*- coding: utf-8 -*-
+"""
+Class for calculating top-up statistics.
+"""
+
+import warnings
+
+from ..utilities.sets import EventsTablesUnion
+from .metaclasses import SubscriberFeature
+
+valid_stats = {"count", "sum", "avg", "max", "min", "median", "stddev", "variance"}
+
+
+class TopUpAmount(SubscriberFeature):
```

Comment:
I think I'd prefer this split into two files, mostly because the second class is massively more complex than the first.